src/App.tsx

```diff
@@ -1,61 +1,124 @@
-import './App.scss';
+import { useState } from 'react';
+import cn from 'classnames';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types';
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [currentTodos, setCurrentTodos] = useState(todosFromServer);
+  const [isEmpty, setIsEmpty] = useState(true);
+  const [isUserSelected, setIsUserSelected] = useState(true);
+
+  const reset = () => {
+    setTitle('');
+    setSelectedUserId(0);
+  };
+
+  const addTodo = () => {
+    const todosId: number[] = currentTodos.map(todo => todo.id);
+
+    const newTodo: Todo = {
+      title,
+      userId: selectedUserId,
+      completed: false,
+      id: Math.max(...todosId) + 1,
+    };
+
+    setCurrentTodos([...currentTodos, newTodo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      setIsEmpty(false);
+    }
+
+    if (!selectedUserId) {
+      setIsUserSelected(false);
+    }
+
+    if (!title || !selectedUserId) {
+      return;
+    }
+
+    addTodo();
+    reset();
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          Title:
+          <input
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter title..."
+            value={title}
+            onChange={event => {
+              setTitle(event.target.value);
+
+              if (!event.target.value) {
+                setIsEmpty(true);
+              }
+            }}
```

Comment:
It's bad practice use anonymous function in JSX
Move logic into separate handler and pass reference here